dashboard/src/modules/components/OverviewComponent/ExpiringSoonComponent.jsx

```diff
@@ -1,19 +1,65 @@
 import "./index.less";
 
+import {
+  DEFAULT_PER_PAGE_EXPIRING,
+  START_PAGE_NUMBER,
+} from "assets/constants/overviewConstants";
 import { List, ListItem } from "@patternfly/react-core";
+import React, { useCallback, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
 
-import React from "react";
-import { useSelector } from "react-redux";
+import { RenderPagination } from "./common-component";
+import { setExpiringRows } from "actions/overviewActions";
 
 const ExpiringSoonComponent = () => {
-  const { expiringRuns } = useSelector((state) => state.overview);
+  const dispatch = useDispatch();
+  const { expiringRuns, initExpiringRuns } = useSelector(
+    (state) => state.overview
+  );
+
+  /* Pagination */
+  const [perPage, setPerPage] = useState(DEFAULT_PER_PAGE_EXPIRING);
+  const [page, setPage] = useState(START_PAGE_NUMBER);
 
+  const onSetPage = useCallback(
+    (_evt, newPage, _perPage, startIdx, endIdx) => {
+      setPage(newPage);
+      dispatch(setExpiringRows(expiringRuns.slice(startIdx, endIdx)));
+    },
+    [dispatch, expiringRuns]
+  );
+  const perPageOptions = [
+    { title: "10", value: 10 },
+    { title: "15", value: 15 },
+    { title: "20", value: 20 },
+  ];
```

Comment:
The default is `DEFAULT_PER_PAGE_EXPIRING` which is `7`, right?  It is weird that the default does not correspond to one of the available choices, and it's even weirder that there is no option to set the value back to the default once it's been changed.

Reply:
Updated